daisy/workflow/workflow.go

```diff
@@ -428,20 +457,33 @@ func (w *Workflow) populateStep(step *Step) error {
 
 func (w *Workflow) populate() error {
 	w.id = randString(5)
-	now := time.Now()
-	datestamp := now.Format("20060102")
-	datetimestamp := now.Format("20060102150405")
-	timestamp := strconv.FormatInt(now.Unix(), 10)
+	now := time.Now().UTC()
+	cu, err := user.Current()
+	if err != nil {
+		return err
+	}
+	w.username = cu.Username
+
+	autovars := map[string]string{
+		"ID":        w.id,
+		"DATE":      now.Format("20060102"),
+		"DATETIME":  now.Format("20060102150405"),
+		"TIMESTAMP": strconv.FormatInt(now.Unix(), 10),
+		"USERNAME":  w.username,
+	}
 
-	// Do replacement from Vars.
 	vars := map[string]string{}
 	for k, v := range w.Vars {
 		vars[k] = v
 	}
+
 	var replacements []string
 	for k, v := range vars {
 		replacements = append(replacements, fmt.Sprintf("${%s}", k), v)
 	}
+	for k, v := range autovars {
```

Comment:
We might want to put this for loop above the previous for loop. Reason? Until now, user vars overrode autovars. With this autovars for loop AFTER the vars for loop, the autovars ID, DATE, DATETIME, TIMESTAMP, and USERNAME **can't** be overridden. That's fine, but inconsistent since the other autovars NAME, ZONE, PROJECT, etc. **can** be overridden.